frontend/server/src/DAO/Users.php

```diff
@@ -447,4 +447,32 @@ final public static function validateDeletionToken(
 
         return boolval($count);
     }
+
+    public static function findByParentalToken(string $token): ?\OmegaUp\DAO\VO\Users {
+        $fields = join(
+            ', ',
+            array_map(
+                fn (string $field): string => "u.{$field}",
+                array_keys(
+                    \OmegaUp\DAO\VO\Users::FIELD_NAMES
+                )
+            )
+        );
+        $sql = "SELECT
+                      {$fields}
+
+                    FROM
+                        Users u
+                    INNER JOIN
+                        Identities i ON u.main_identity_id = i.identity_id
+                    WHERE
+                      parental_verification_token = ?
+                    LIMIT 1';";
+
+         /** @var int|null */
+        return \OmegaUp\MySQLConnection::getInstance()->GetOne(
```

Comment:
This function should return the whole object for User, so you can use the function `GetRow` instead of `GetOne`
```suggestion
        $result = \OmegaUp\MySQLConnection::getInstance()->GetRow(
```

Reply:
okay!! I have returned whole object for User.